Pyrado/pyrado/algorithms/meta/sprl.py

```diff
@@ -0,0 +1,573 @@
+        """
+        self._mean = mean.clone().detach().requires_grad_(True)
+        self._cov_chol_flat = cov_chol_flat.clone().detach().requires_grad_(True)
+        self.distribution = MultivariateNormal(self.mean, self.cov)
+
+    @staticmethod
+    def from_stacked(dim: int, stacked: np.ndarray) -> "MultivariateNormalWrapper":
+        r"""
+        Creates an instance of this class from the given stacked numpy array as generated e.g. by
+        `MultivariateNormalWrapper.get_stacked(self)`.
+
+        :param dim: dimensionality `k` of the random variable
+        :param stacked: array containing the mean and standard deviations of shape `(2 * k,)`, where the first `k`
+                        entries are the mean and the last `k` entries are the standard deviations
+        :return: a `MultivariateNormalWrapper` with the given mean/cov.
+        """
+        if not (len(stacked.shape) == 1):
+            raise pyrado.ValueErr(msg="Stacked has invalid shape! Must be 1-dimensional.")
+        if not (stacked.shape[0] == 2 * dim):
+            raise pyrado.ValueErr(
+                msg="Stacked has invalid size! Must be 2*dim (one times for mean, a second time for covariance cholesky diagonal)."
+            )
+
+        mean = stacked[:dim]
+        cov_chol_flat = stacked[dim:]
+
+        return MultivariateNormalWrapper(to.tensor(mean).double(), to.tensor(cov_chol_flat).double())
+
+    @property
+    def dim(self):
+        """ Gets the size (dimensionality) of the random variable. """
+        return self._mean.shape[0]
+
+    @property
+    def mean(self):
+        """ Gets the mean. """
+        return self._mean
+
+    @mean.setter
+    def mean(self, mean: to.Tensor):
+        """ Sets the mean. """
+        if not (mean.shape == self.mean.shape):
+            raise pyrado.ShapeErr(given_name="mean", expected_match=self.mean.shape)
+
+        self._mean = mean
+        self._update_distribution()
+
+    @property
+    def cov(self):
+        """ Gets the covariance matrix, shape `(k, k)`. """
+        return (self._cov_chol_flat ** 2).diag()
+
+    @property
+    def cov_chol_flat(self):
+        """
+        Gets the standard deviations, i.e. the diagonal entries of the Cholesky decomposition of the covariance matrix,
+        shape `(k,)`.
+        """
+        return self._cov_chol_flat
+
+    @cov_chol_flat.setter
+    def cov_chol_flat(self, cov_chol_flat: to.Tensor):
+        """ Sets the standard deviations, shape `(k,)`. """
+        if not (cov_chol_flat.shape == self.cov_chol_flat.shape):
+            raise pyrado.ShapeErr(given_name="cov_chol_flat", expected_match=self.cov_chol_flat.shape)
+
+        self._cov_chol_flat = cov_chol_flat
+        self._update_distribution()
+
+    def parameters(self) -> List[to.Tensor]:
+        """ Gets the parameters (mean and standard deviations) of this distribution. """
+        return [self.mean, self.cov_chol_flat]
```

Comment:
This function reminds me of torch.nn.Modules.parameters() which returns a generator. How about you do it the same way. As far as I see, you are only using it once later (actually to use it liek a generator). Establishing this parallelism would be nice.

Reply:
Done in bcdc32aa4a8d02ea06ca04b85098b97bb60c1447